var/spack/repos/builtin/packages/matlab/package.py

```diff
@@ -45,8 +45,19 @@ class Matlab(Package):
 
     version('R2016b', 'b0e0b688894282139fa787b5a86a5cf7')
 
-    variant('mode', default='interactive', description='Installation mode (interactive, silent, or automated)')
-    variant('key',  default='',            description='The file installation key to use')
+    variant(
+        'mode',
+        default='interactive',
+        values=('interactive', 'silent', 'automated'),
+        description='Installation mode (interactive, silent, or automated)'
+    )
+
+    variant(
+        'key',
+        default='',
+        values=lambda x: True,  # Anything goes as a key
```

Comment:
Commit comment: "	variants: fixed packages reported by @adamjstewart in #4098" I'll add it to the description.